### PR TITLE
Submit dependabot dependency graph to Github

### PR DESCRIPTION
**Pull request by mdedetrich**
This PR uses scala centers official integration of sbt with github's dependabot (see https://github.com/scalacenter/sbt-dependency-submission) to submit any potential security vulnerabilities. This is a more practical solution compared to https://github.com/apache/incubator-pekko/pull/289 especially in regards to getting it ready for release candidate.

https://github.com/aiven/guardian-for-apache-kafka/security/dependabot is a nice demonstration of what it looks like when it finds security vulnerabilities.

### CODE REVIEW

**Review by nvollmar — APPROVED** [review]
lgtm

**Review by gmethvin — APPROVED** [review]

**Comment by mdedetrich** [comment]
@pjfanning The PR has been approved but I will wait for your approval incase you have any comments

**Comment by raboof** [comment]
> https://github.com/aiven/guardian-for-apache-kafka/security/dependabot is a nice demonstration of what it looks like when it finds security vulnerabilities.

(this page is only visible if you have permissions for that repo)

**Comment by mdedetrich** [comment]
> > https://github.com/aiven/guardian-for-apache-kafka/security/dependabot is a nice demonstration of what it looks like when it finds security vulnerabilities.
> 
> (this page is only visible if you have permissions for that repo)

Thanks, it seems like `.asf.yml` already supports this so I enabled it right now, force pushed.

**Comment by mdedetrich** [comment]
I am going to merge the PR fow now so we can see if there are any potential issues before release, can always revert if there is a problem./

**Comment by pjfanning** [comment]
We have 100% false negatives in https://github.com/apache/incubator-pekko/security/dependabot. Not one of the jackson-databind is applicable. It appears that whatever is going, dependabot is not aware that we are using jackson 2.14.3. All the issues that dependabot highlights are fixed in that version.

**Comment by mdedetrich** [comment]
> We have 100% false negatives in https://github.com/apache/incubator-pekko/security/dependabot. Not one of the jackson-databind is applicable. It appears that whatever is going, dependabot is not aware that we are using jackson 2.14.3. All the issues that dependabot highlights are fixed in that version.
> 
> Some alerts relate to org.codehaus.jackson:jackson-mapper-asl but `sbt dependencyTree` does not show any such dependency.

Looking into this now, seems to only be happening with Pekko. Under the hood the workflow is an sbt plugin that resolves the dependency graph so it must be doing something odd.

**Comment by mdedetrich** [comment]
Issue made at https://github.com/apache/incubator-pekko/issues/376

**Pull request merged**
